### ISSUE TITLE
Picom makes Krita's canvas stutter, but xcompmgr (or no compositor at all) does not

### ISSUE THREAD

**Issue by Pragmageddon8** [yshui/compton#1154]

**Comment by Monsterovich** [comment]
Try `--no-frame-pacing`

**Comment by le-dino** [comment]
> Try `--no-frame-pacing`

unfortunately didn't make a difference, it still stutters

**Comment by Monsterovich** [comment]
@le-dino I suppose you have `ForceFullCompositionPipeline` and `ForceCompositionPipeline` options disabled?

**Comment by absolutelynothelix** [comment]
can't reproduce. the only stuttering that i have is because of frame pacing and it could be fixed either with `--no-frame-pacing` on the command line or with `no-frame-pacing = true;` in the configuration file (frame pacing doesn't work well on nvidia gpus now but it should be fixed soon).

**Comment by le-dino** [comment]
> @le-dino I suppose you have `ForceFullCompositionPipeline` and `ForceCompositionPipeline` options disabled?

no, the tablet runs on 60hz but I have `ForceCompositionPipeline` enabled because without it, horrible screen tearing is introduced.
Just to make sure, I tested with full composition enabled too, but the issue persists.

> can't reproduce. the only stuttering that i have is because of frame pacing and it could be fixed either with `--no-frame-pacing` on the command line or with `no-frame-pacing = true;` in the configuration file (frame pacing doesn't work well on nvidia gpus now but it should be fixed soon).

I tried both methods, but it does not make it better for me.

**Comment by Monsterovich** [comment]
@le-dino 

> no, the tablet runs on 60hz but I have `ForceCompositionPipeline` enabled because without it, horrible screen tearing is introduced. Just to make sure, I tested with full composition enabled too, but the issue persists.

These settings should be disabled if you are using a compositor, because ForceCompositionPipeline is a **compositor**, but implemented within the video driver. V-Sync in picom should be enabled to prevent screen tearing.

**Comment by le-dino** [comment]
> These settings should be disabled if you are using a compositor, because ForceCompositionPipeline is a **compositor**, but implemented within the video driver. V-Sync in picom should be enabled to prevent screen tearing.

I understand.
Disabled all composition pipelines on all my monitors, `vsync = true;`. There is still very bad screen tearing, due to this I cannot tell if it's any smoother.

Tested `no-frame-pacing = true;`, `backend = "xr_glx_hybrid";`, `backend = "xrender";`, `backend = "glx";`;
no visible change.

**Comment by Monsterovich** [comment]
> > These settings should be disabled if you are using a compositor, because ForceCompositionPipeline is a **compositor**, but implemented within the video driver. V-Sync in picom should be enabled to prevent screen tearing.
> 
> I understand. Disabled all composition pipelines on all my monitors, `vsync = true;`. There is still very bad screen tearing, due to this I cannot tell if it's any smoother.
> 
> Tested `no-frame-pacing = true;`, `backend = "xr_glx_hybrid";`, `backend = "xrender";`, `backend = "glx";`; no visible change.

I have an RTX 3060 Ti with V-sync enabled in picom (glx backend) and don't have the issues you have. I just installed Krita via Flatpak and checked it out.

[2023-12-22_21-30-43.mkv.zip](https://github.com/yshui/picom/files/13756272/2023-12-22_21-30-43.mkv.zip)

Recorded video at 60fps even though I have a 144hz monitor.

It is possible that the problem is in the driver 545.29.06 because I'm using 535.129.03?

**Comment by le-dino** [comment]
Upon further inspection, I found out another thing. I don't know if it's useful information, but while having picom enabled the performance in videogames is degraded. This isn't necessarily a problem since I can simply disable the compositor while gaming and it doesn't cause any problems, but it just so happens that xcompmgr doesn't cause any problems. It's always smooth and seamless, never requiring me to disable it at all.

So I suppose this isn't a krita only issue after all, but I'm confused because it's like xcompmgr has no impact on anything at all, while being enabled, but picom degrades the performance of everything. Which I don't understand, since it's based off xcompmgr.
